src/cmd/deploy.ts

```diff
@@ -548,12 +563,94 @@ export default class DefenderDeploy {
     );
   }
 
+  private async deployCategories(output: DeployOutput<DefenderCategory>) {
+    const categories: YCategory[] = this.serverless.service.resources?.Resources?.categories ?? [];
+    const client = getSentinelClient(this.teamKey!);
+    const notifications = await client.listNotificationChannels();
+    const retrieveExisting = () => client.listNotificationCategories();
+
+    await this.wrapper<YCategory, DefenderCategory>(
+      this.serverless,
+      'Categories',
+      categories,
+      retrieveExisting,
+      // on update
+      async (category: YCategory, match: DefenderCategory) => {
+        const newCategory = constructNotificationCategory(
+          this.serverless,
+          category,
+          match.stackResourceId!,
+          notifications,
+        );
+        const mappedMatch = {
+          name: match.name,
+          description: match.description,
+          notificationIds: match.notificationIds,
+          stackResourceId: match.stackResourceId,
+        };
+        if (_.isEqual(validateTypesAndSanitise(newCategory), validateTypesAndSanitise(mappedMatch))) {
+          return {
+            name: match.stackResourceId!,
+            id: match.categoryId,
+            success: false,
+            response: match,
+            notice: `Skipped ${match.stackResourceId} - no changes detected`,
+          };
+        }
+        // Warn users when they try to change the category name
+        if (match.name !== category.name) {
```

Comment:
Might make sense to throw here and deliver an informative error message (and I suppose maybe this would be part of the client package and fairly certain I just approved the PR). Silly example, but what if they were changing "Low" to "Critical" (because they only have Medium/High/Critical) and changed the notification IDs. If we ignore a name change, we might set their critical channels to low

Reply:
Not allowing the `name` change is enforced in the API, so not too worried about that. The v0 scope defines only 3 categories, this will change later on, at which point we will enable the create/delete endpoints as well, to allow for more customisation.

As for throwing vs warning, I can see pros and cons to both approaches. Currently we take the `warn` approach for all changes that are not allowed / possible, e.g. sentinel network or type changes. The idea behind it was whether its worth interrupting the entire deploy process for this, what do you think?

Reply:
> Not allowing the name change is enforced in the API

Does enforced means it throws or ignores? Where else do we take the approach that an invalid request to update a param ignores? I thought at least in some of those cases we throw

Reply:
> Does enforced means it throws or ignores?
At the moment it ignores the `name` field, so essentially only allows to update the description and notification channels for that category.

I don't have a strong opinion on whether we should throw or warn in this case though. Your call.

Reply:
ok, I think changing the name is a bad request given current implementation and should throw (though this should be done on the server side, don't think there are needed changes here) - but would need to be messaged appropriately if users try to do it here